client/src/main/java/org/apache/rocketmq/client/impl/ClientRemotingProcessor.java

```diff
@@ -213,4 +221,79 @@ private RemotingCommand consumeMessageDirectly(ChannelHandlerContext ctx,
 
         return response;
     }
+
+    private RemotingCommand receiveReplyMssage(ChannelHandlerContext ctx,
+        RemotingCommand request) throws RemotingCommandException {
+
+        final RemotingCommand response = RemotingCommand.createResponseCommand(null);
+        long receiveTime = System.currentTimeMillis();
+        ReplyMessageRequestHeader requestHeader = (ReplyMessageRequestHeader) request.decodeCommandCustomHeader(ReplyMessageRequestHeader.class);
+
+        try {
+            MessageExt msg = new MessageExt();
+            msg.setTopic(requestHeader.getTopic());
+            msg.setQueueId(requestHeader.getQueueId());
+            msg.setStoreTimestamp(requestHeader.getStoreTimestamp());
+
+            if (requestHeader.getBornHost() != null) {
+                String[] bornHostArr = requestHeader.getBornHost().split("/");
+                String bornHost/*ip:port*/ = bornHostArr[bornHostArr.length - 1];
+                String[] host = bornHost.split(":");
+                if (host.length == 2)
+                    msg.setBornHost(new InetSocketAddress(host[0], Integer.parseInt(host[1])));
+            }
+
+            if (requestHeader.getStoreHost() != null) {
+                String[] storeHostArr = requestHeader.getStoreHost().split("/");
```

Comment:
Maybe need to consider IPV6(such as: 2408:4004:0180:8100:3faa:1dde:2b3f:898a), more details can be referred in org.apache.rocketmq.common.sysflag.MessageSysFlag#BORNHOST_V6_FLAG and org.apache.rocketmq.store.CommitLog#putMessage